fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/logical/LogicalHudiScan.java

```diff
@@ -0,0 +1,218 @@
+import org.apache.doris.nereids.properties.LogicalProperties;
+import org.apache.doris.nereids.trees.TableSample;
+import org.apache.doris.nereids.trees.expressions.ComparisonPredicate;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.GreaterThan;
+import org.apache.doris.nereids.trees.expressions.GreaterThanEqual;
+import org.apache.doris.nereids.trees.expressions.LessThanEqual;
+import org.apache.doris.nereids.trees.expressions.Slot;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.literal.StringLiteral;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.RelationId;
+import org.apache.doris.nereids.trees.plans.visitor.PlanVisitor;
+import org.apache.doris.nereids.util.Utils;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Logical Hudi scan for Hudi table
+ */
+public class LogicalHudiScan extends LogicalFileScan {
+    private static final Logger LOG = LogManager.getLogger(LogicalHudiScan.class);
+
+    // for hudi incremental read
+    private Optional<TableScanParams> scanParams;
+    private Optional<IncrementalRelation> incrementalRelation;
+
+    /**
+     * Constructor for LogicalHudiScan.
+     */
+    protected LogicalHudiScan(RelationId id, ExternalTable table, List<String> qualifier,
+            Optional<GroupExpression> groupExpression, Optional<LogicalProperties> logicalProperties,
+            Set<Expression> conjuncts, SelectedPartitions selectedPartitions, Optional<TableSample> tableSample,
+            Optional<TableSnapshot> tableSnapshot,
+            Optional<TableScanParams> scanParams, Optional<IncrementalRelation> incrementalRelation) {
+        super(id, table, qualifier, groupExpression, logicalProperties, conjuncts,
+                selectedPartitions, tableSample, tableSnapshot);
+        this.scanParams = scanParams;
+        this.incrementalRelation = incrementalRelation;
+    }
+
+    public LogicalHudiScan(RelationId id, ExternalTable table, List<String> qualifier,
+            Optional<TableSample> tableSample, Optional<TableSnapshot> tableSnapshot) {
+        this(id, table, qualifier, Optional.empty(), Optional.empty(),
+                Sets.newHashSet(), SelectedPartitions.NOT_PRUNED, tableSample, tableSnapshot,
+                Optional.empty(), Optional.empty());
+    }
+
+    /**
+     * Set scan params for incremental read
+     *
+     * @param table
+     * @param scanParams
+     */
+    public void setScanParams(HMSExternalTable table, TableScanParams scanParams) {
```

Comment:
should not do set in Nereids, if u want to update some attr, u should impl withXXX and return a new object